src/corecel/Macros.hh

```diff
@@ -216,3 +216,27 @@
             }                                              \
         } while (0),                                       \
         HANDLE_EXCEPTION)
+
+/*!
+ * \def CELER_DEFAULT_COPY_MOVE
+ *
+ * Explicitly declares defaulted copy and move constructors ans assignment
+ * operators.
+ */
+#define CELER_DEFAULT_COPY_MOVE(CLS)      \
+    CLS(CLS const&) = default;            \
+    CLS& operator=(CLS const&) = default; \
+    CLS(CLS&&) = default;                 \
+    CLS& operator=(CLS&&) = default;
```

Comment:
I think we might want to leave off the trailing `;` since all of our other statement-like macros require them. See https://github.com/celeritas-project/celeritas/pull/851#discussion_r1261759708 . I will make a follow-on PR.